app/models/request.rb

```diff
@@ -0,0 +1,31 @@
+# == Schema Information
+#
+# Table name: requests
+#
+#  id         :integer          not null, primary key
+#  student_id :integer
+#  state      :integer          default(0)
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Request < ActiveRecord::Base
+  validates :student_id, presence: true
+
+  has_many :course_requests
```

Comment:
if you want to use `course_requests` as a join table, you should use `has_many :courses through: :course_requests`. then you can do something like `request.courses`